nemo/collections/common/data/lhotse/dataloader.py

```diff
@@ -276,3 +281,18 @@ def _normalize_loudness(cuts: CutSet, db_norm: float) -> CutSet:
 
 def _merge_supervisions(cuts: CutSet) -> CutSet:
     return cuts.merge_supervisions()
+
+
+def _flatten_paired_text(cut) -> list:
+    ans = []
+    # For multiple paired texts for single input. Flattens cut into multiple cuts.
+    if not isinstance(cut.custom["text"], dict):
+        return [cut]
+    cut = cut.move_to_memory(audio_format="wav")  # performs I/O once and holds audio in memory from now on
+    # Popping to ease eyesight on debug
+    paired_text = cut.custom.pop("text")
+    for lang, data in paired_text.items():
+        text_instance = cut.map_supervisions(lambda s: fastcopy(s, text=data["text"], language=lang))
+        text_instance.custom = data
```

Comment:
@pzelasko How well to dict methods map to the LasyJsonlIterator. I'm noticing some randomness if i use update here. Sometimes the custom attributes won't transfer to the new cut.

Reply:
`fastcopy` performs a (very) shallow copy, it's possible `cut.custom` remains shared between instances, and when you pop stuff it disappears everywhere. To be 100% safe you may want to keep a copy `custom_cpy = cut.custom.copy()` and then in new cut copies do sth like `new_cut.custom = {**custom_cpy, "override_arg": "override_val", ...}`

Reply:
Well the issue is the new instance's `custom` is changing. So despite being assigned to data, it looks like it gets overriden in the lazy copying.